com.unity.render-pipelines.universal/Documentation~/rendering/deferred-rendering-path.md

```diff
@@ -0,0 +1,422 @@
+
+Examples of such shaders:
+
+* **Complex Lit**: the Lighting model of this shader (for example, the Clear Coat effect) is too complex and extra Material properties cannot fit into the G-buffer.
+
+* **Baked Lit** and **Unlit**: these shaders do not receive dynamic lighting, so writing them directly into the Lighting buffer instead of the G-buffer is faster.
+
+* **Custom shaders**: Unity renders the shaders that do not declare the Pass tags required by the Deferred Rendering Path as Forward-only. The required Pass tags are: `LightMode`, and `UniversalMaterialType`. For more information, see [URP Pass tags](../urp-shaders/urp-shaderlab-pass-tags.md).
+
+Unity renders Materials with such shaders in the Forward Rendering Path. For the SSAO Renderer Feature to be able to calculate ambient occlusion for the Materials using the **Complex Lit** shader, Unity must render such Materials in the depth and normal prepass first. This is because Unity does not render those Materials in the G-buffer pass (GBufferPass). For more information, see [URP Pass tags](../urp-shaders/urp-shaderlab-pass-tags.md).
+
+#### General implementation notes
+
+For maximum platform compatibility, the URP Deferred Rendering Path uses the light stencil volume technique to render light volumes and apply deferred shading.
+
+## Relevant code files
+
+This section contains the list of files that contain the code related to the Deferred Rendering Path.
+
+* The main class that handles the Deferred Rendering Path:
+
+    ```
+    com.unity.render-pipelines.universal\Runtime\DeferredLights.cs
+    ```
+
+* ScriptableRenderPass for the G-Buffer pass:
+
+    ```
+    com.unity.render-pipelines.universal\Runtime\Passes\GBufferPass.cs
+    ```
+
+* ScriptableRenderPass for the deferred shading pass:
+
+    ```
+    com.unity.render-pipelines.universal\Runtime\Passes\DeferredPass.cs
+    ```
+
+* Shader asset for the deferred shading:
+
+    ```
+    com.unity.render-pipelines.universal\Shaders\Utils\StencilDeferred.shader
+    ```
+
+* Utility functions for the deferred shading:
+
+    ```
+    com.unity.render-pipelines.universal\Shaders\Utils\Deferred.hlsl
+    ```
+
+* Utility functions for storing and loading the Material properties from the G-buffer:
+
+    ```
+    com.unity.render-pipelines.universal\Shaders\Utils\UnityGBuffer.hlsl
+    ```
+
+## ShaderLab Pass tags
+
+To enable Unity to render a shader in the Deferred Rendering Path, the shader must have a Pass with the following tag definition:
+
+`"LightMode" = "UniversalGBuffer"`
+
+Unity executes the shader with such `LightMode` tag during the G-buffer Pass.
+
+To indicate that Unity must render a certain Material in the Forward-only Pass in the Deferred Rendering Path, add the following tags to a shader Pass:
+
+`"LightMode" = "UniversalForwardOnly"`
+
+`"LightMode" = "DepthNormalsOnly"`
+
+To specify the shader lighting model (Lit, SimpleLit), use the `UniversalMaterialType` tag.
```

Comment:
Maybe this could be expanded a little but to talk about all the possible `UniversalMaterialTypes`?

Reply:
Thank you, I've added this to a separate task:
https://unity3d.atlassian.net/browse/DOC-3262